src/api/api_routes.py

```diff
@@ -0,0 +1,68 @@
+# Non-public API. Don't host publicly - SECURITY RISKS!
+# (will only be on with --api starting option)
+# Currently no API stability guarantees are provided - API may break on any new commit.
+
+from fastapi import FastAPI, Body
+from fastapi.exceptions import HTTPException
+from itertools import tee
+
+import gradio as gr
+
+from typing import Dict, List
+
+from src.common_constants import GenerationOptions as go
+from src.misc import SCRIPT_VERSION
+from src.api.api_constants import api_options, models_to_index
+from src.api.api_core import api_gen, encode_to_base64
+
+# gr.Blocks is needed for auto1111 extensions
+def depth_api(_: gr.Blocks, app: FastAPI):
+    @app.get("/depth/version")
+    async def version():
+        return {"version": SCRIPT_VERSION}
+
+    @app.get("/depth/get_options")
+    async def get_options():
+        return {
+            "gen_options": [x.name.lower() for x in go],
+            "api_options": api_options,
+            "model_names": models_to_index.keys()
+        }
+
+    @app.post("/depth/generate")
+    async def process(
+        input_images: List[str] = Body([], title='Input Images'),
+        generate_options: Dict[str, object] = Body({}, title='Generation options', options= [x.name.lower() for x in go]),
+        api_options: Dict[str, object] = Body({'outputs': ["depth"]}, title='Api options', options= api_options)
```

Comment:
I have used the old depth api quite a bit. It it very frustrating how many extra pictures it sends once you start using options like rembg or clipping. There really does need to be an option to specify what actually want sent back because often it does lead to unnecessary overhead client-side. 

This might not be the exact right approach. You're right that it should be optional.